src/System.Globalization/tests/CultureInfo/CultureInfoCalendar.cs

```diff
@@ -2,19 +2,50 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
 using Xunit;
 
 namespace System.Globalization.Tests
 {
     public class CultureInfoCalendar
     {
+        public static IEnumerable<object[]> Calendar_TestData()
+        {
+            yield return new object[] { CultureInfo.InvariantCulture, typeof(GregorianCalendar) };
+            yield return new object[] { new CultureInfo("en-US"), typeof(GregorianCalendar) };
+            yield return new object[] { new CultureInfo("th-TH"), typeof(ThaiBuddhistCalendar) };
+            yield return new object[] { new CultureInfo("ja-JP"), typeof(GregorianCalendar) };
+            yield return new object[] { new CultureInfo("ar-SA"), typeof(UmAlQuraCalendar) };
+        }
+
         [Theory]
-        [InlineData("en-US", typeof(GregorianCalendar))]
-        [InlineData("th-TH", typeof(ThaiBuddhistCalendar))]
-        public void Calendar(string locale, Type expected)
+        [MemberData(nameof(Calendar_TestData))]
+        public void Calendar(CultureInfo cultureInfo, Type expected)
         {
-            CultureInfo cultureInfo = new CultureInfo(locale);
             Assert.IsType(expected, cultureInfo.Calendar);
+            Assert.Same(cultureInfo.Calendar, cultureInfo.Calendar);
```

Comment:
again this is kind of hardcoding values. every while we encounter countries change their default calendar. I would suggest to test if the culture support this calendar instead of testing this calendar is the default calendar